src/bin/pg_autoctl/cli_do_tmux.c

```diff
@@ -0,0 +1,441 @@
+ * script buffer, with an additional Enter command.
+ */
+static void
+tmux_add_send_keys_command(PQExpBuffer script, const char *fmt, ...)
+{
+	char buffer[BUFSIZE] = { 0 };
+	va_list args;
+
+	va_start(args, fmt);
+	pg_vsprintf(buffer, fmt, args);
+	va_end(args);
+
+	appendPQExpBuffer(script, "send-keys '%s' Enter\n", buffer);
+}
+
+
+/*
+ * tmux_add_xdg_environment appends the XDG environment that makes the test
+ * target self-contained, as a series of tmux send-keys commands, to the given
+ * script buffer.
+ */
+static void
+tmux_add_xdg_environment(PQExpBuffer script, const char *root)
+{
+	char *xdg[][3] = {
+		{ "XDG_DATA_HOME", "share" },
+		{ "XDG_CONFIG_HOME", "config" },
+		{ "XDG_RUNTIME_DIR", "run" },
+	};
+
+	/*
+	 * For demo/tests purposes, arrange a self-contained setup where everything
+	 * is to be found in the given options.root directory.
+	 */
+	for (int i = 0; i < 3; i++)
+	{
+		char *var = xdg[i][0];
+		char *dir = xdg[i][1];
+
+		tmux_add_send_keys_command(script,
+								   "export %s=\"%s/%s\"", var, root, dir);
+	}
+}
+
+
+/*
+ * tmux_pg_autoctl_create appends a pg_autoctl create command to the given
+ * script buffer, and also the commands to set PGDATA and PGPORT.
+ */
+static void
+tmux_pg_autoctl_create(PQExpBuffer script,
+					   const char *root,
+					   int pgport,
+					   const char *role,
+					   const char *name)
+{
+	char *pg_ctl_opts = "--hostname localhost --ssl-self-signed --auth trust";
+
+	tmux_add_xdg_environment(script, root);
+	tmux_add_send_keys_command(script, "export PGPORT=%d", pgport);
+
+	/* the monitor is always named monitor, and does not need --monitor */
+	if (strcmp(role, "monitor") == 0)
+	{
+		tmux_add_send_keys_command(script, "export PGDATA=\"%s/monitor\"", root);
+
+		tmux_add_send_keys_command(script,
+								   "%s create %s %s --run",
+								   pg_autoctl_argv0,
+								   role,
+								   pg_ctl_opts);
+	}
+	else
+	{
+		char monitor[BUFSIZE] = { 0 };
+
+		sformat(monitor, sizeof(monitor),
+				"$(%s show uri --pgdata %s/monitor --monitor)",
+				pg_autoctl_argv0,
+				root);
+
+		tmux_add_send_keys_command(script,
+								   "export PGDATA=\"%s/%s\"",
+								   root,
+								   name);
+
+		tmux_add_send_keys_command(script,
+								   "%s create %s %s --monitor %s --run",
+								   pg_autoctl_argv0,
+								   role,
+								   pg_ctl_opts,
+								   monitor);
+	}
+}
+
+
+/*
+ * keeper_cli_tmux_script generates a tmux script to run a test case or a demo
+ * for pg_auto_failover easily.
+ */
+void
+cli_do_tmux_script(int argc, char **argv)
+{
+	TmuxOptions options = tmuxOptions;
+
+	char *root = options.root;
+	PQExpBuffer script = createPQExpBuffer();
+
+	int pgport = options.firstPort;
+
+	if (script == NULL)
+	{
+		log_error("Failed to allocate memory");
+		exit(EXIT_CODE_INTERNAL_ERROR);
+	}
+
+	tmux_add_command(script, "set-option -g default-shell /bin/bash");
+	tmux_add_command(script, "new -s pgautofailover");
```

Comment:
This session name should include the root I think, right now you cannot start two clusters at the same time.

Reply:
The root could be any path on-disk. I've used the first pgport instead, which is known to be a number with not too many digits.